python/cudf/cudf/core/buffer.py

```diff
@@ -171,62 +247,34 @@ def deserialize(cls, header: dict, frames: list) -> Buffer:
 
         return buf
 
-    @classmethod
-    def empty(cls, size: int) -> Buffer:
-        return Buffer(rmm.DeviceBuffer(size=size))
+    def __repr__(self) -> str:
+        return (
+            f"<cudf.core.buffer.Buffer size={format_bytes(self._size)} "
+            f"ptr={hex(self._ptr)} owner={repr(self._owner)} "
+        )
 
-    def copy(self) -> Buffer:
-        """
-        Create a new Buffer containing a copy of the data contained
-        in this Buffer.
-        """
-        from rmm._lib.device_buffer import copy_device_to_ptr
-
-        out = Buffer.empty(size=self.size)
-        copy_device_to_ptr(self.ptr, out.ptr, self.size)
-        return out
 
-
-def _buffer_data_from_array_interface(array_interface):
-    ptr = array_interface["data"][0]
-    if ptr is None:
-        ptr = 0
-    itemsize = cudf.dtype(array_interface["typestr"]).itemsize
-    shape = (
-        array_interface["shape"] if len(array_interface["shape"]) > 0 else (1,)
-    )
-    size = functools.reduce(operator.mul, shape)
-    return ptr, size * itemsize
-
-
-def confirm_1d_contiguous(array_interface):
-    strides = array_interface["strides"]
-    shape = array_interface["shape"]
-    itemsize = cudf.dtype(array_interface["typestr"]).itemsize
-    typestr = array_interface["typestr"]
-    if typestr not in ("|i1", "|u1", "|b1"):
-        raise TypeError("Buffer data must be of uint8 type")
-    if not get_c_contiguity(shape, strides, itemsize):
-        raise ValueError("Buffer data must be 1D C-contiguous")
-
-
-def get_c_contiguity(shape, strides, itemsize):
-    """
-    Determine if combination of array parameters represents a
-    c-contiguous array.
+def get_ptr_and_size(array_interface: Mapping) -> Tuple[int, int]:
     """
-    ndim = len(shape)
-    assert strides is None or ndim == len(strides)
+    Return the pointer and size of an array interface.
 
-    if ndim == 0 or strides is None or (ndim == 1 and strides[0] == itemsize):
-        return True
+    Raises ValueError if array isn't C-contiguous
+    """
 
-    # any dimension zero, trivial case
-    for dim in shape:
-        if dim == 0:
+    def is_c_contiguous(shape, strides, itemsize):
```

Comment:
any reason this is now defined inside a parent function? It could be useful outside of this context someday (although, clearly I assumed that long ago and it has not happened yet :) )

Reply:
Nope, moved `is_c_contiguous()` to module level :)